client/client-lib/src/lib.rs

```diff
@@ -243,6 +243,21 @@ impl<T: AsRef<ClientConfig> + Clone> Client<T> {
         Ok(OutPoint { txid, out_idx: 0 })
     }
 
+    /// Validate tokens without claiming them. This function checks if signatures are valid
+    /// based on the federation public key. It does not check if the nonce is unspent.
+    pub async fn validate_tokens(&self, coins: Coins<SpendableCoin>) -> Result<()> {
```

Comment:
I returned `Result<()>` but wasn't sure what type this should return

Reply:
Returning `Result<()>` here is the right way. What you should do though is using a reference for the coins like so: `coins: &Coins<SpendableCoin>` so callers don't need to unnecessarily clone the struct.

Reply:
Thanks for the headsup. I revised the function definition to use a reference to coins